azext_iot/digitaltwins/providers/resource.py

```diff
@@ -230,6 +253,168 @@ def remove_role(self, name, assignee, role_type=None, resource_group_name=None):
             dt_scope=target_instance.id, assignee=assignee, role_type=role_type
         )
 
+    # Identity
+
+    def assign_identity(
+        self,
+        name,
+        system_identity=None,
+        user_identities=None,
+        identity_role=None,
+        identity_scopes=None,
+        resource_group_name=None
+    ):
+        target_instance = self.find_instance(
+            name=name, resource_group_name=resource_group_name
+        )
+
+        if bool(identity_role) ^ bool(identity_scopes):
+            raise RequiredArgumentMissingError(
+                'At least one scope (--scopes) and one role (--role) required for system-managed identity role assignment.'
+            )
+        if not system_identity and not user_identities:
+            raise RequiredArgumentMissingError(
+                'No identities provided to assign. Please provide system (--system) or user-assigned identities (--user).'
+            )
+
+        if not target_instance.identity:
+            target_instance.identity = DigitalTwinsIdentity()
+
+        if user_identities:
+            if not target_instance.identity.user_assigned_identities:
+                target_instance.identity.user_assigned_identities = {}
+            for user_identity in user_identities:
+                identity = target_instance.identity.user_assigned_identities.get(user_identity, {})
+                target_instance.identity.user_assigned_identities[user_identity] = identity
+
+        has_system_identity = target_instance.identity.type in [
+            IdentityType.system_assigned_user_assigned.value, IdentityType.system_assigned.value
+        ]
+
+        if system_identity or has_system_identity:
+            if target_instance.identity.user_assigned_identities:
+                target_instance.identity.type = IdentityType.system_assigned_user_assigned.value
+            else:
+                target_instance.identity.type = IdentityType.system_assigned.value
+        else:
+            if target_instance.identity.user_assigned_identities:
+                target_instance.identity.type = IdentityType.user_assigned.value
+            else:
+                target_instance.identity.type = IdentityType.none.value
+
+        try:
+            update_poller = self.mgmt_sdk.digital_twins.create_or_update(
+                resource_name=name,
+                resource_group_name=resource_group_name,
+                digital_twins_create=target_instance,
+                long_running_operation_timeout=60,
+            )
+
+            def rbac_handler(lro):
```

Comment:
Isn't this rbac_handler a copy of the one for instance create? Can we reduce duplication?

Reply:
as long as you are fine with partial functions

Reply:
sure